module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,324 @@
+                        }
+                        else {
+                            incrementKeyValues[`result.${singleResponse.qid}.${singleResponse.value}`] = 1;
+                        }
+                    })
+
+                    let updateQuery = { 
+                        $inc: incrementKeyValues
+                    }
+
+                    await pollsHelper.updatePollDocument(pollId, updateQuery)
+            
+                    return resolve({
+                        success: true,
+                        message: messageConstants.apiResponses.POLL_SUBMITTED,
+                        data: true
+                    });
+
+                } else {
+                   throw new Error(messageConstants.apiResponses.POLL_COULD_NOT_BE_SUBMITTED)
+                }
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+   static list(userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if(userId == ""){
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+            }
+
+            let getPollList = [
+                pollsHelper.list
+                ( 
+                    userId
+                ),
+                database.models.pollSubmissions.aggregate
+                    (
+                        [
+                            { "$match": { "userId": userId, isDeleted: false } },
+                            {
+                                "$project": {
+                                    '_id': "$pollId",
+                                    'name': '$pollName'
+                                }
+                            },
+                            { "$sort": { createdAt: 1 } }
+                        ])
+            ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+                          
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                          result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                          result = result.concat(response[1]);
+                        }
+                    });
+
+
+                if (!result.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: _.uniqWith(result, _.isEqual)
```

Comment:
@deepakr123 - How does this isEqual work - https://lodash.com/docs/4.17.15#isEqual as per the link it has to be passed 2 arguments?

Reply:
resolved